specification/security/resource-manager/Microsoft.Security/preview/2023-05-01-preview/healthReports.json

```diff
@@ -0,0 +1,363 @@
+      "required": [
+        "issueKey"
+      ],
+      "properties": {
+        "issueKey": {
+          "description": "The unique issue key",
+          "type": "string"
+        },
+        "issueName": {
+          "type": "string",
+          "description": "The issue name"
+        },
+        "securityValues": {
+          "type": "array",
+          "description": "The affected security values that MDC offers that will be affected by the issue, for example: recommendations, alerts, etc",
+          "items": {
+            "type": "string",
+            "description": "security values"
+          }
+        },
+        "issueDescription": {
+          "type": "string",
+          "description": "The issue description"
+        },
+        "remediationSteps": {
+          "type": "string",
+          "description": "Human readable description of what you should do to mitigate this health issue"
+        },
+        "remediationScript": {
+          "type": "string",
+          "description": "The remediation script to solve this issue"
+        },
+        "issueAdditionalData": {
+          "type": "object",
+          "description": "Additional data for the given issue. The additional data depends on the issue type",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "environmentDetails": {
+      "type": "object",
+      "description": "The environment details of the resource",
+      "properties": {
+        "nativeResourceId": {
+          "description": "The native resource id of the resource (in case of Azure - the resource Id, in case of MC - the native resource id)",
+          "type": "string"
+        },
+        "environmentHierarchyId": {
+          "description": "The hierarchy id of the connector (in case of Azure - the subscription Id, in case of MC - the hierarchyId id)",
+          "type": "string"
+        },
+        "organizationalHierarchyId": {
+          "description": "The organizational hierarchy id of the connector (in case of Azure - the subscription Id, in case of MC - the organizational hierarchyId id)",
+          "type": "string"
+        },
+        "subscriptionId": {
+          "description": "The subscription Id",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "The tenant Id",
+          "type": "string"
+        }
+      }
+    },
+    "healthDataClassification": {
+      "type": "object",
+      "description": "The classification of the health report",
+      "properties": {
+        "component": {
+          "type": "string",
+          "description": "The component describes the name of the agent/service that scans the issue"
+        },
+        "scenario": {
+          "type": "string",
+          "description": "The scenario describes the health scenario issue of the component"
+        },
+        "scope": {
+          "type": "string",
```

Comment:
Did this previously get modeled as enum? Why change it?

Reply:
Yes. We decided with our product manager that it is better as a string. There will be a lot of new values for this field, and we don't want to create a new API version each time.

Reply:
Will the customer need to know what are all the possible values they may need to plan for when e.g. writing their code to process them? How will they find out?

Reply:
@roni93 This is the part that looks like a cross-version breaking change to me. If not in the REST API payload, at least in the contact, because you're saying that your new version can return any string where previously it only returned a few.

Reply:
@TimLovellSmith got you.
Can you please guide me on what should I do/or with who I need to talk?

Reply:
In some ways it looks like you've kind of planned for this to happen by providing the 'Unknown' value which you can return to old api-version callers. But I'm still not sure why you are changing it to string from enum because that seems like it is probably going to be an SDK breaking change, where the old constant values for those enum values get removed etc. @raych1 could you chime in on whether this is OK?